x/concentrated-liquidity/math/math.go

```diff
@@ -73,13 +77,19 @@ func CalcAmount0Delta(liq, sqrtPriceA, sqrtPriceB osmomath.BigDec, roundUp bool)
 		// - calculating amountIn during swap
 		// - adding liquidity (request user to provide more tokens in in favor of the pool)
 		// The denominator is truncated to get a higher final amount.
-		return liq.MulRoundUp(diff).QuoRoundUp(sqrtPriceA).QuoRoundUp(sqrtPriceB).Ceil()
+		// Note that the order of divisions is important here. First, we divide by a larger number (sqrtPriceB) and then by a smaller number (sqrtPriceA).
+		// This leads to a smaller error amplification.
+		// TODO (perf): QuoRoundUpMut with no reallocation.
+		return liq.MulRoundUp(diff).QuoRoundUp(sqrtPriceB).QuoRoundUp(sqrtPriceA).Ceil()
```

Comment:
Hmm it's not immediately obvious to me why this should reduce error across the board – would you be able to clarify?

It seems, for instance, that whether the spot prices are below 1 should make a difference here since then the division will increase the result, so at least we should expect the optimal division order to flip at this boundary.

Reply:
You're right that this is only applicable assuming that one of the sqrt prices is below 1. Expanded on this in the comment

Reply:
[41e436c](https://github.com/osmosis-labs/osmosis/pull/6369/commits/41e436ce8db2082ed359967ab2ddd46b9557e5ab)